master/buildbot/steps/shell.py

```diff
@@ -112,7 +112,7 @@ def __init__(self, workdir=None,
         buildstep_kwargs = {}
         # workdir is here first positional argument, but it belongs to BuildStep parent
         kwargs['workdir'] = workdir
-        for k in kwargs.keys()[:]:
+        for k in list(kwargs)[:]:
```

Comment:
It makes a copy, but so does list ... this doc says the two are equivalent for lists:
https://docs.python.org/2/library/functions.html#list